### ISSUE TITLE
Design: OADP NFS plugin for full NFS support

### ISSUE THREAD

**Issue by weshayutin** [openshift/oadp-operator#1045]

**Comment by sseago** [comment]
Right. For a while, we've said that we just need a filesystem object store plugin, but to do that we'll have to resolve the download request issue -- i.e. anything backed by an object store plugin needs to allow for velero to provide the client a URL to download the logs, tarball, etc. Basically anywhere that a client works (velero pod, user laptop, etc.), that client needs to be able to download these backup/restore artifacts via the URL retrieved from the download request.

**Comment by weshayutin** [comment]
https://issues.redhat.com/browse/RHSTOR-3545

**Comment by sseago** [comment]
@weshayutin Are there any details about OCS NFS support that make it relevant to this task? What's needed is to implement the object store API for an (out-of-cluster) NFS back end, and to figure out how to make download requests work with NFS (possibly requiring some sort of additional controller, on top of a plugin). I don't know that OCS supporting in-cluster NFS helps us with either one of these.
